lib/WebService/MyAffiliates.pm

```diff
@@ -45,25 +45,48 @@ sub __ua {
 }
 
 ## https://myaffiliates.atlassian.net/wiki/display/PUB/Feed+1%3A+Users+Feed
-sub get_users {            ## no critic (ArgUnpacking)
+sub get_users {    ## no critic (ArgUnpacking)
     my $self = shift;
     my %args = @_ % 2 ? %{$_[0]} : @_;
     my $url  = Mojo::URL->new('/feeds.php?FEED_ID=1');
     $url->query(\%args) if %args;
     return $self->request($url->to_string);
 }
 
+sub create_affiliate {
+    my $self = shift;
+    my $args = shift;
+    croak 'A hashref was expected as parameter' unless ref $args eq ref {};
```

Comment:
This check isn't needed, the next line will die

Reply:
I think is better to be explicit about the error.